CHANGELOG.md

```diff
@@ -37,6 +37,7 @@ in 3.17. Use `PaymentSettings.defaultTransactionFlowStrategy` instead.
 ### Other changes
 - Fix error in variant available stock calculation - 13593 by @awaisdar001
 - Fix giftcard code description - #13728 by @rafiwts
+- Add a new `page` field on `AssignedPageAttributeValue`. First stage (migration) of a simplification of Attribute - Page relation from #13403. by michal-macioszczyk
```

Comment:
Are you aware that merging this to main, means that the data migration will happen in 3.16. It means that you will be able to use it in 3.17. So the refactor logic that you're going to introduce will need to be delivered to 3.17
As far as I understand, it currently looks like this:
- main (future 3.16) received changes from this PR
- 3.16 release and migration data, trigger the celery task to update the attributes
- <wait until it ends>
- 3.17 receives the logic that can be sure that all "new" attribute  values are already propagated

I would say that this PR should be added to 3.15 (as a patch release). Then your main logic can be delivered to 3.16.
So the flow would look:
- 3.15.X received a data migration
- celery task updates the attributes
- (main) 3.16 recieve (+ update PR that was merged to 3.15*) the logic that can be sure that all "new" attribute values are already propagated

* The celery task introduced in 3.15, should be converted to standard migration in 3.16

Reply:
We discussed that outside of github so I'll just add for visibility. For both this PR and #13778 we're aiming at adding the task in 3.16 and the rest in 3.17. So this is similar to the above strategy just moved by 1 version 👯